code_dir/src/my_lib/decode.rs

```diff
@@ -0,0 +1,114 @@
+use crate::my_lib::levenshtein_functions::{Action, DeltaAction};
+use memmap2::Mmap;
+use std::fs::File;
+use std::io::Write;
+
+pub(crate) fn decode_file_with_chunks(input_path: &str, output_path: &str) {
+    let input = File::open(input_path).expect("file not open");
+    let mut output = File::create(output_path).expect("file not create");
+
+    let memory_map = unsafe { Mmap::map(&input).expect("Failed to create memory map") };
+    let mut offset = 0usize;
+    let header_size = 9usize;
+    let max_offset = input.metadata().unwrap().len() as usize;
+
+    while offset < max_offset {
+        let header_chunk = &memory_map[offset..offset + header_size];
+        match header_chunk[0] {
+            0 => {
+                let mut length_arr = [0u8; 8];
+                for index_length in 0..8 {
+                    length_arr[index_length] = header_chunk[index_length + 1];
+                }
+                let length =
+                    unsafe { std::mem::transmute::<[u8; 8], u64>(length_arr) }.to_le() as usize;
```

Comment:
```suggestion
               let length = u64::from_le_bytes(length_arr) as usize;
```
https://stackoverflow.com/questions/71542843/convert-bytes-to-u64